File: android/app/src/main/java/com/controle/financeiro/CustomWebViewClient.java
package com.controle.financeiro;

import android.webkit.WebResourceError;
import android.webkit.WebResourceRequest;
import android.webkit.WebView;
import android.webkit.WebViewClient;

public class CustomWebViewClient extends WebViewClient {
    @Override
    public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request) {
        // Handle all URLs within the WebView
        view.loadUrl(request.getUrl().toString());
        return true;
    }

    @Override
    public void onReceivedError(WebView view, WebResourceRequest request, WebResourceError error) {
        super.onReceivedError(view, request, error);
        // Handle errors here
        view.loadUrl("file:///android_asset/www/error.html");
    }

    @Override
    public void onPageFinished(WebView view, String url) {
        super.onPageFinished(view, url);
        // Page loading finished
    }
}

File: android/app/src/main/kotlin/com/controle/financeiro/CustomWebViewClient.kt
package com.controle.financeiro

import android.graphics.Bitmap
import android.webkit.WebResourceError
import android.webkit.WebResourceRequest
import android.webkit.WebView
import android.webkit.WebViewClient
import androidx.webkit.WebViewClientCompat

class CustomWebViewClient : WebViewClientCompat() {
    
    override fun shouldOverrideUrlLoading(view: WebView, request: WebResourceRequest): Boolean {
        // Handle all URLs within the WebView
        view.loadUrl(request.url.toString())
        return true
    }

    override fun onPageStarted(view: WebView, url: String, favicon: Bitmap?) {
        super.onPageStarted(view, url, favicon)
        // Show loading indicator if needed
    }

    override fun onPageFinished(view: WebView, url: String) {
        super.onPageFinished(view, url)
        // Hide loading indicator if needed
    }

    override fun onReceivedError(
        view: WebView,
        request: WebResourceRequest,
        error: WebResourceError
    ) {
        super.onReceivedError(view, request, error)
        // Load error page when there's a problem
        if (request.isForMainFrame) {
            view.loadUrl("file:///android_asset/www/error.html")
        }
    }

    override fun shouldInterceptRequest(
        view: WebView,
        request: WebResourceRequest
    ) = super.shouldInterceptRequest(view, request)

    override fun onReceivedHttpError(
        view: WebView,
        request: WebResourceRequest,
        errorResponse: android.webkit.WebResourceResponse
    ) {
        super.onReceivedHttpError(view, request, errorResponse)
        // Handle HTTP errors if needed
        if (request.isForMainFrame) {
            view.loadUrl("file:///android_asset/www/error.html")
        }
    }
}
